app/static/src/app/components/WodinPanels.vue

```diff
@@ -33,19 +46,38 @@ import { computed, defineComponent, ref } from "vue";
 import VueFeather from "vue-feather";
 
 export enum PanelsMode {
-        Left, Right, Both
-    }
+    Left, Right, Both
+}
+
+export enum HidePanelContent {
+    Left, Right, None
+}
+
+enum DragStart {
+    Icon, Edge
+}
+
+type WidthStyle = {
+    width?: string
+}
 
 export default defineComponent({
     name: "WodinPanels",
     components: {
         VueFeather
     },
     setup() {
-        const mode = ref<PanelsMode>(PanelsMode.Both);
-        const leftAnimateClass = ref("");
-        const rightAnimateClass = ref("");
+        const windowWidth = computed(() => window.innerWidth);
+        // different snap tolerance because the plot does not look
+        // good with width of windowWidth / 8
+        const widthToleranceLeft = windowWidth.value / 8;
+        const widthToleranceRight = windowWidth.value / 4;
+        const windowBoundaryLeft = widthToleranceLeft;
+        const windowBoundaryRight = windowWidth.value - widthToleranceRight;
+        const snapToleranceLeft = windowWidth.value / 25;
+        const snapToleranceRight = windowWidth.value - windowWidth.value / 13;
```

Comment:
Can we have comments about what all these things actually mean? widthTolerance is the minimum width at which the panel content is shown, and snapTolerance is the width at which the panel being dragged will snap to 0?

I think we discussed that these proportions seem a bit arbitrary - if the point is to hide content that is mangled by small widths, it might be more obvious to have absolute pixel tolerances - that would lead to issues at the smallest widths though, so I think this is fine!

Reply:
yep can add comments, it is arbitrary but it also works fairly nicely, the /25 is completely arbitrary, the /13 is lower because there is some inherent padding in the view charts collapsed mode that isnt there in the view options collapse mode.

I think the best way to handle these small bits is maybe we keep the threshold like `Math.max(windowBoundaryLeft, 200)` for example and that way we can handle larger screens and smaller ones!